### PR TITLE
Supplemental Claims | Refactor code with va-link

### PR DESCRIPTION
**Pull request by rocio-desantiago**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > Updated web component with Va component
- _(If bug, how to reproduce)_
  > Not a bug
- _(What is the solution, why is this the solution)_
  > To reflect VA design system as shown [here](https://design.va.gov/storybook/?path=/docs/components-va-link--default#download), we are implementing the va component for the download link.
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits Team 1 | Benefits decision reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#60030

## Testing done

- _Describe what the old behavior was prior to the change_
  > same behavior
- _Describe the steps required to verify your changes are working as expected_
  > verify that it works the same with a test user (no visual changes)
     -  Go to this [link](https://staging.va.gov/decision-reviews/supplemental-claim/file-supplemental-claim-form-20-0995/start) and select "Another type of claim (not a disability claim)" for the "What type of claim are you filing a Supplemental Claim for?". Click Continue.
     - The following page should be this [one](https://staging.va.gov/decision-reviews/supplemental-claim/file-supplemental-claim-form-20-0995/start) where Download Link is present.
- _Describe the tests completed and the results_

  > **Unit tests updated:** 
  > - subtask.unit.spec.jsx
  > - basic-2-page.unit.spec.jsx
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

|    -------   | va-link | 
| ------- | ------ |
| Before | download-link-before | 
| After | va-link |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by rocio-desantiago** [comment]
@Mottie is it a design requirement for the file size to be included in the download link as shown in the before photo? `Download VA Form 20-0995 PDF (1.7MB)`

**Comment by Mottie** [comment]
The web component [download `va-link`](https://design.va.gov/storybook/?path=/docs/components-va-link--download#download) only supports adding `pages`, not megabyte sizes.

**Review by Mottie — APPROVED** [review]

**Review by powellkerry — APPROVED** [review]

File: src/platform/forms/tests/sub-task/basic-2-page.unit.spec.jsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ describe('the Supplemental Claims Sub-task', () => {
 
     fireEvent.click($('va-button[continue]', container), mouseClick);
     expect($('form[data-page="other"]', container)).to.exist;
-    expect($('a[download]', container)).to.exist;
+    expect($('va-link', container)).to.exist;
```
Comment (Mottie) [inline src/platform/forms/tests/sub-task/basic-2-page.unit.spec.jsx:68]: It might be better to be more specific here, like `va-link[download]`

Comment (rocio-desantiago) [reply src/platform/forms/tests/sub-task/basic-2-page.unit.spec.jsx]: @Mottie good call. Thank you!

**Pull request merged**